protocol/streaming/full_node_streaming_manager.go

```diff
@@ -796,6 +840,42 @@ func (sm *FullNodeStreamingManagerImpl) GetSubaccountSnapshotsForInitStreams(
 	return ret
 }
 
+func (sm *FullNodeStreamingManagerImpl) addBatchUpdatesToCache(
```

Comment:
Docstring comment for when this is used / what exactly the batching does?

Reply:
might be good to add a comments saying this should never be called without a lock since we're `streamUpdateCache`